### DISCUSSION TITLE
Fallback per path?

### DISCUSSION THREAD

**Discussion by eLvErDe** [bluenviron/mediamtx#315] [Questions]
Hello,

First I would like to thank you for releasing this RTSP proxy, it seems to be working just fine and will help me lot with my projects.

Then a quick question:
Is it possible to have a fallback onDemand source for each path ?
The use case is the following: each path has an onDemand camera, if the stream is not available a fallback is provide using closest camera.

Does it makes sense ?

Best regards, Adam.

**Comment by aler9** [comment]
Hello, fallback and sourceOnDemand can't be used together.

This is what happens when VLC requires a sourceOnDemand:

1. VLC connects to the server
2. the server connects to the camera
3. VLC times out before the camera and closes the connection
4. the camera times out (5 secs)
5. the server tries to send the fallback URL, but it's too late, since VLC has already closed the connection 

Therefore, fallback can be used only when sourceOnDemand is `no`

[a comment by an automated account is not shown]
